server/data/hmppsAuthClient.ts

```diff
@@ -71,14 +83,23 @@ export default class HmppsAuthClient {
     return this.restClient(token).get({ path: '/api/user/me' }) as Promise<User>
   }
 
-  getUserByEmailAddress(token: string, emailAddress: string): Promise<User> {
-    logger.info(`Getting user detail by email address: calling HMPPS Auth`)
-    return this.restClient(token).get({ path: `/api/authuser`, query: { email: emailAddress } }) as Promise<User>
+  async getUserByEmailAddress(token: string, emailAddress: string): Promise<AuthUser> {
+    logger.info(`Getting auth user detail by email address: calling HMPPS Auth`)
+    const authUsers = (await this.restClient(token).get({
+      path: `/api/authuser`,
+      query: { email: emailAddress },
+    })) as AuthUser[]
+
+    return authUsers[0]
```

Comment:
what happens when the API returns an empty list ?

Reply:
when this occurs the response code is 204 success. so it won't throw an exception.

Reply:
We've fixed this now